File: src/main/java/com/example/bongabdodemo/controller/IndexController.java
package com.example.bongabdodemo.controller;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.Calendar;

import static com.example.bongabdodemo.utils.BongabdoCalculation.*;

@Controller
@RequestMapping("")
public class IndexController {
    @GetMapping
    public String indexPage(Model model) {
        model.addAttribute("gregorianDate", getDateString(Calendar.getInstance()));
        return "index";
    }

    @PostMapping
    public String convert(@RequestParam("gregorianDate") String gregorianDate, Model model) {
        model.addAttribute("gregorianDate", gregorianDate);

        var gregorianDateInCalendar = getCalendarFromString(gregorianDate);
        model.addAttribute("banglaAcademyOutput", getBanglaAcademyBongabdo(gregorianDateInCalendar));
        model.addAttribute("drikShiddhantaOutput", getDrikShiddhantaBongabdo(gregorianDateInCalendar));
        model.addAttribute("conversionAvailable", true);
        return "index";
    }
}

File: src/main/kotlin/com/example/bongabdodemo/controller/IndexController.kt
package com.example.bongabdodemo.controller

import com.example.bongabdodemo.utils.BongabdoCalculation.getBanglaAcademyBongabdo
import com.example.bongabdodemo.utils.BongabdoCalculation.getCalendarFromString
import com.example.bongabdodemo.utils.BongabdoCalculation.getDateString
import com.example.bongabdodemo.utils.BongabdoCalculation.getDrikShiddhantaBongabdo
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import java.util.*

@Controller
@RequestMapping(path = [""])
class IndexController {
    @GetMapping
    fun indexPage(model: Model): String {
        model.addAttribute("gregorianDate", getDateString(Calendar.getInstance()));
        return "index"
    }

    @PostMapping
    fun convert(@RequestParam("gregorianDate") gregorianDate: String, model: Model): String {
        model.addAttribute("gregorianDate", gregorianDate);

        val gregorianDateInCalendar = getCalendarFromString(gregorianDate)
        model.addAttribute("banglaAcademyOutput", getBanglaAcademyBongabdo(gregorianDateInCalendar));
        model.addAttribute("drikShiddhantaOutput", getDrikShiddhantaBongabdo(gregorianDateInCalendar));
        model.addAttribute("conversionAvailable", true);
        return "index";
    }
}